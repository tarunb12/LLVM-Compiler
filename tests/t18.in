// Modular inverse

int modularInverse(int k, int p) {
    int n = k % p
    int i
    for (i = 0; i < p; i++) {
        if ((n * i) % p == 1) return i
    }
    return -1
}

int main() {
    int n = 8
    int p = 37
    int i
    for (i = 1; i <= n; i++) {
        printf("Modular inverse of %d (mod %d): %d\n", i, p, modularInverse(i, p))
    }
    return 0
}