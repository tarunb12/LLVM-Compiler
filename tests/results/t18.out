Modular inverse of 1 (mod 37): 1
Modular inverse of 2 (mod 37): 19
Modular inverse of 3 (mod 37): 25
Modular inverse of 4 (mod 37): 28
Modular inverse of 5 (mod 37): 15
Modular inverse of 6 (mod 37): 31
Modular inverse of 7 (mod 37): 16
Modular inverse of 8 (mod 37): 14
